api/ttn/lorawan/v3/user.proto

```diff
@@ -553,6 +553,26 @@ message ListUserBookmarksRequest {
 
   // Only return recently deleted bookmarks.
   bool deleted = 5;
+
+  // Specifies that only bookmarks that reference the given entity type should be returned. If empty the filter is not
+  // applied.
+  repeated string has_entity_type = 6 [(validate.rules).repeated = {
```

Comment:
Just `entity_types`, and please limit the number of elements to at most 6.

Reply:
will change the name but doesn't the `unique: true` restriction already limits the possible amount of values to be 6?

Reply:
You are correct, and indeed `unique` is better than the length check.

Reply:
did the change regarding the naming and removal of the `""` value withing the allowed values of the array.

> will change the name but doesn't the unique: true restriction already limits the possible amount of values to be 6?

I suppose the proper question would be if there is a reason to add the maximum size restriction on top of the unique restriction.

Reply:
There should not be in my view.